tracer/src/Datadog.Tracer.Native/iast/module_info.cpp

```diff
@@ -0,0 +1,842 @@
+    {
+        return GetFieldInfo(token);
+    }
+    else if (typeFromToken == mdtMethodSpec)
+    {
+        return GetMethodSpec(token);
+    }
+    return NULL;
+}
+MethodInfo* ModuleInfo::GetMethodInfo(mdMethodDef methodDef)
+{
+    if (methodDef == 0x06000000)
+    {
+        return NULL; // Wrong methodDef
+    }
+    CSGUARD(_cs);
+    auto it = _methods.find(methodDef);
+    if (it != _methods.end())
+    {
+        return it->second;
+    }
+    MethodInfo* res = new MethodInfo(this, methodDef);
+    _methods[methodDef] = res;
+    return res;
+}
+FieldInfo* ModuleInfo::GetFieldInfo(mdFieldDef fieldDef)
+{
+    CSGUARD(_cs);
+    auto it = _fields.find(fieldDef);
+    if (it != _fields.end())
+    {
+        return it->second;
+    }
+    FieldInfo* res = new FieldInfo(this, fieldDef);
+    _fields[fieldDef] = res;
+    return res;
+}
+MethodSpec* ModuleInfo::GetMethodSpec(mdMethodSpec methodSpec)
+{
+    CSGUARD(_cs);
+    auto it = _specs.find(methodSpec);
+    if (it != _specs.end())
+    {
+        return it->second;
+    }
+    MethodSpec* res = new MethodSpec(this, methodSpec);
+    _specs[methodSpec] = res;
+    return res;
+}
+SignatureInfo* ModuleInfo::GetSignature(mdSignature signatureToken)
+{
+    CSGUARD(_cs);
+    auto it = _signatures.find(signatureToken);
+    if (it != _signatures.end())
+    {
+        return it->second;
+    }
+    PCCOR_SIGNATURE pSig;
+    ULONG nSig;
+    if (SUCCEEDED(this->_metadataImport->GetSigFromToken(signatureToken, &pSig, &nSig)))
+    {
+        SignatureInfo* res = new SignatureInfo(this, pSig, nSig);
+        _signatures[signatureToken] = res;
+        return res;
+    }
+    return NULL;
+}
+
+bool ModuleInfo::AreSameTypes(mdTypeRef typeRef1, mdTypeRef typeRef2)
+{
+    if (typeRef1 == typeRef2)
+    {
+        return true;
+    }
+    HRESULT hr = S_OK;
+    WCHAR typeName1[1024];
+    WCHAR typeName2[1024];
+    hr = _metadataImport->GetTypeRefProps(typeRef1, NULL, typeName1, 1024, NULL);
+    if (SUCCEEDED(hr))
+    {
+        hr = _metadataImport->GetTypeRefProps(typeRef2, NULL, typeName2, 1024, NULL);
+    }
+    if (SUCCEEDED(hr))
+    {
+        WSTRING t1(typeName1);
+        WSTRING t2(typeName2);
+        return t1 == t2;
+    }
+    return false;
+}
+
+HRESULT ModuleInfo::GetAssemblyTypeRef(const WSTRING& assemblyName, const WSTRING& typeName, mdTypeRef* typeRef)
+{
+    HRESULT hr = S_OK;
+
+    mdAssemblyRef assemblyRef;
+    IfFailRet(GetAssemblyRef(assemblyName, &assemblyRef));
+    IfFailRet(GetTypeRef(assemblyRef, typeName, typeRef));
+    return hr;
+}
+
+std::vector<MethodInfo*> ModuleInfo::GetMethods(mdTypeDef typeDef)
+{
+    HCORENUM hCorEnum = nullptr;
+    std::vector<MethodInfo*> methods;
+    mdMethodDef enumeratedMethods[128];
+    ULONG enumCount;
+
+    HRESULT hr = this->_metadataImport->EnumMethods(
+        &hCorEnum, typeDef, enumeratedMethods, sizeof(enumeratedMethods) / sizeof(enumeratedMethods[0]), &enumCount);
+    if (SUCCEEDED(hr))
+    {
```

Comment:
you can call `methods.reserve(enumCount)` here to avoid intermediate allocations

Reply:
Fixed